pkg/server/serveroption/option.go

```diff
@@ -26,14 +44,63 @@ var Default = Options{
 type Options struct {
 	// Both sizes should really be uint32 but it's better saved for a single
 	// refactor commit.
-
 	MaxFlows      int // max number of flows that can be stored in the ring buffer
 	MonitorBuffer int // buffer size for monitor payload
+
+	CiliumDaemon CiliumDaemon // when running inside Cilium, contains a reference to the daemon
+
+	OnServerInit   []OnServerInit   // invoked when the hubble server is initialized
+	OnMonitorEvent []OnMonitorEvent // invoked before an event is decoded
+	OnDecodedFlow  []OnDecodedFlow  // invoked after a flow has been decoded
 }
 
+// returning `stop: true` from a callback stops the execution chain, regardless
+// of the error encountered (for example, explicitly filtering out certain
+// events, or similar).
+type stop = bool
```

Comment:
I didn't know one could define [type alias](https://golang.org/ref/spec#Type_declarations) instead of just type definitions! Neat and handy.

Reply:
The thanks goes to @glibsm for this (he wrote this original code). :)

Reply:
golang 1.9 was eagerly anticipated by library authors for this very reason. This allowed much better backwards compatibility in addition to better API definitions that spelled out what the simple type was used for